lib/util/viewer.js

```diff
@@ -425,10 +430,12 @@ export function stopIsFlex(stopData) {
  * The RouteRenderer requires a leg. To get it to render a Route object,
  * we use this method
  */
-export function generateFakeLegForRouteRenderer(route) {
+export function generateFakeLegForRouteRenderer(route, hideTopBorder = false) {
   // RouteRenderer requires a "leg"
   return {
     agencyName: route?.agencyName || route.agency?.name,
+    // Don't render top border when colors are mismatched
+    hideTopBorder,
```

Comment:
If you create different kinds of route renderers, each can decide whether to render a border or not.

Reply:
I think I see what you mean now. But wouldn't this wrapper route renderer still wind up passing the same prop to the underlying route renderer? I agree this prop is poorly named and should instead be used to send context information to the route renderer. Maybe something like `isBeingRenderedInCompactSpace` or similar.

Reply:
In my mind, each combination of visual styles (borders, how color is applied, shape...) would be a different renderer.

Regarding adjusting route names to the available space, one technique would be to render text as SVG, then the SVG would be fitted in the space available. See for instance https://github.com/opentripplanner/otp-ui/blob/47302686e8bb53256d3214c1cf1d54db7ebdf754/packages/transit-vehicle-overlay/src/VehicleIcon.tsx#L30 in https://github.com/opentripplanner/otp-ui/pull/470.

Reply:
So if the config is so that route background colors should be used in `PatternRow`, then use one kind of renderer. If the config sets a say white background color to `PatternRow`, use another kind of renderer.

Reply:
Also for the SVG trick -- are we sure that this doesn't have a11y implications?

Reply:
Correct, an 1`aria-label` is needed if using SVG.

Reply:
Going back to route renderers, to me, the sample below features four different route name renderers.
<img width="213" alt="image" src="https://user-images.githubusercontent.com/56846598/201173134-1d99cd1b-1c94-432c-8a81-3796cda501da.png">